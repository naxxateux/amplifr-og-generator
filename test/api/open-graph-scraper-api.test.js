const openGraphScraper = jest.requireActual('open-graph-scraper')

async function askScraper (page) {
  let options = {
    url: page,
    headers: {
      'user-agent':
        'facebookexternalhit/1.1 (+http://www.facebook.com/externalhit_uatext.php)'
    },
    timeout: 10000,
    ogImageFallback: false,
    onlyGetOpenGraphInfo: false
  }

  return openGraphScraper(options)
}

it('checks response format from Open Graph Scraper', async () => {
  let result = await askScraper('https://amplifr.com')
  expect(result.result).toMatchSnapshot()
})
