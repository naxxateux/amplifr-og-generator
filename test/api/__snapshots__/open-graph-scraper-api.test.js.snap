// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks response format from Open Graph Scraper 1`] = `
Object {
  "charset": "utf8",
  "ogDescription": "Amplifr is a social media publication and analytics app that analyses conversion and revenue for each of your social media updates. Learn how many purchases, button clicks, newsletter signups or other goal conversions you had for your social media marketing efforts!",
  "ogImage": Object {
    "height": null,
    "type": "png",
    "url": "https://amplifr.com/share-en.png",
    "width": null,
  },
  "ogLocale": "en",
  "ogSiteName": "Amplifr",
  "ogTitle": "Amplifr: Social media scheduling with social media ROI analytics",
  "ogType": "website",
  "ogUrl": "https://amplifr.com/en/",
  "requestUrl": "https://amplifr.com",
  "success": true,
  "twitterCard": "summary_large_image",
  "twitterCreator": "@amplifr",
  "twitterSite": "@amplifr",
}
`;
